cabal.project

```diff
@@ -1,4 +1,5 @@
 packages: .
+          semantic-analysis
```

Comment:
We’re starting to get a lot of packages in this hierarchy. I don’t think this is a bad thing—indeed, I think this will help us with compile times and tooling reliability—but we should start documenting in the README our notions of how the project is organized, and which responsibilities each subproject covers.

Reply:
Yeah, @dcreager and I have talked about pulling (some of) the architecture diagram I drew up into this repo.

Was also thinking we might want to have a languages subdir. Not sure yet. Thoughts?